apps/sync/js/adapters/bookmarks.js

```diff
@@ -51,12 +52,30 @@ var BookmarksHelper = (() => {
     });
   }
 
+  function removeSyncedCollectionMtime(userid) {
+    return new Promise(resolve =>
+      asyncStorage.removeItem(userid + BOOKMARKS_COLLECTION_MTIME, resolve));
+  }
+
   function getSyncedCollectionMtime(userid) {
     return new Promise(resolve => {
       asyncStorage.getItem(userid + BOOKMARKS_COLLECTION_MTIME, resolve);
     });
   }
 
+  function getLastRevisionId(userid) {
```

Comment:
Sorry, forgot to comment. I renamed `checkIfCleared` to `checkIfClearedSince` and renamed the variable returned to `newRevisionId`, to distinguish it from `lastRevisionId`. I also moved the call to `getLastRevisionId` from `checkIfClearedSince` to `handleClear`. We could merge `checkIfClearedSince` into `handleClear`, but it's quite a big function. Maybe I didn't understand your comment, will ping you on irc!